### ISSUE TITLE
Feature Request: White Color on darker Colors tuning

### ISSUE THREAD

**Issue by tiagomsmagalhaes** [johnpapa/vscode-peacock#109]
# White Text/Icon color on darker colors tuning

macOS 10.13.6
VS Code 1.32.3 (1.32.3) stable

~~Would be nice to detect the current color and decide to display white color for text and icons. It may be more noticeable the noticeability (pun intended) of the information on the status bar.~~

I have chosen for whatever reason this color `#498aff` for one window on VS Code, and it's displaying black text color on the status bar where should be better suited as white. So I suppose that the turning point to whether display black or white text (and icons) could be refined.

**Comment by johnpapa** [comment]
@musicfuel may have some ideas here. We are using the tinycolor library which handles a lot of this contrast for us. It would be helpful to gather a list of color combos that are not as desirable in the product today, so we can try to tune this a bit.

**Comment by musicfuel** [comment]
**TL;DR;** This is probably better handled as a personal preference manual override. We could also consider reintroducing the light and dark foreground manual settings.

We are using [tinycolor](https://github.com/bgrins/TinyColor) which provides some basic color theory mechanisms to determine whether or not to show a light or dark foreground color based on the perceived brightness of the background. More or less, if we think the color is darker than 50% we show the light foreground. If it is greater than 50% we show the dark foreground.

Brightness is measured on a scale of 0-255 where a value of 127.5 is perfectly 50%. Considering you have a color of `#498aff`, this is more or less what is occurring:

```js
const lightForeground = '#e7e7e7';
const darkForegound = '#15202b';
const background = '#498aff';

const perceivedBrightness = tinycolor(background).getBrightness(); // 131.903, so 51.7%
const isDark = tinycolor(background).isDark(); // false, since brightness is above 50%
const textColor = isDark ? lightForeground : darkForeground; // We end up using dark text
```

This particular color is very near 50% on the perceived brightness, but the determination is binary so the color is either light or dark based on which side of 50% it is (exactly 50% is considered as light by the library). For your particular color, `#498aff`, all of the theory aspects that tinycolor provides show that using the dark foreground is the right approach.

```js
const readability = tinycolor.readability(darkForeground, background); // 4.996713
const isReadabled = tinycolor.isReadable(darkForeground, background); // true
```

The readability calculations and metrics are based on [Web Content Accessibility Guidelines (Version 2.0)](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#contrast-ratiodef) and in general a ratio close to 5 is considered good based on that information. In fact, if I run the lightForeground through the same algorithm you can see that readability actually suffers with a reduced contrast ratio

```js
const readability = tinycolor.readability(lightForeground, background); // 2.669008
const isReadabled = tinycolor.isReadable(lightForeground, background); // false
```

All of this is to say that we are using the library to the best of its ability and everything checks out based on that, so this may be better handled as a personal preference. If light text in the status bar is preferred after the color is applied, it can easily be changed manually in `.vscode/settings.json`. To achieve what you want, after using Peacock to apply the color you can manually open that file and change only the `statusBar.foreground` to either `#e7e7e7` or a light color of your choice (note that all the settings with the exception of `statusBar.foreground` came from what Peacock applies in its default configuration):

```js
"workbench.colorCustomizations": {
    "activityBar.background": "#7cabff",
    "activityBar.foreground": "#15202b",
    "activityBar.inactiveForeground": "#15202b99",
    "activityBarBadge.background": "#df0050",
    "activityBarBadge.foreground": "#e7e7e7",
    "titleBar.activeBackground": "#498aff",
    "titleBar.inactiveBackground": "#498aff99",
    "titleBar.activeForeground": "#15202b",
    "titleBar.inactiveForeground": "#15202b99",
    "statusBar.background": "#498aff",
    "statusBarItem.hoverBackground": "#1669ff",
    "statusBar.foreground": "#e7e7e7"
}
```

Another option would be to reintroduce settings that earlier versions of Peacock had to better control the light and the dark foreground colors that Peacock uses. They are currently hardcoded, but having those would allow the user to better control the exact colors used or potentially even reverse them (i.e. purposefully set the Dark Foreground color to a near-white color to get the behavior you want).

@johnpapa Thoughts?

**Comment by johnpapa** [comment]
Yeah - and I think if we really need to handle those cases that are around 50%, then we need to let the user override it themselves. So maybe we need a setting for it. Easy to add.

I welcome feedback on this.

@musicfuel - side note - can you add that fantastic description to the README? It's great for the FAQ

**Comment by johnpapa** [comment]
addressed in 1.4.0

**Comment by johnpapa** [comment]
I also added the explanation to the FAQ

**Comment by johnpapa** [comment]
re-opening ... this will land in v2.0.0

will close when it lands

**Comment by johnpapa** [comment]
closed by #117
